### DISCUSSION TITLE
How can I use customName as a function in transformImport?

### DISCUSSION THREAD

**Discussion by khangviet1996** [web-infra-dev/rsbuild#3747] [Q&A]
After migrating my large CRA project to rsbuild, I noticed significant reductions in both build time and development time. However, there's a project issue I’d like to resolve.

In my project, We're using a barrel file to import common components, as shown here:

`import { Form, FormAddButton } from 'common/components';`

I want to convert it to individual imports, like:

`import FormAddButton from 'common/components/button/FormAddButton';`

I tried using this configuration to achieve that:

```
{
  libraryName: 'common/components',
  customName: 'common/components/button/{{ member }}',
  transformToDefaultImport: true,
}
```
Is there a way to make customName function-based? That would make it much more flexible! Like this!
```
      {
        libraryName: 'common/components',
        customName: (name) => {
          // Assume certain components are in 'form', 'button', and 'notifications'
          if (name === 'Form') return `common/components/form/Form`;
          if (name === 'FormAddButton') return `common/components/button/FormAddButton`;
          if (name === 'CustomNotification') return `common/components/notifications/CustomNotification`;
          return `common/components/others/${name}`;
        },
        transformToDefaultImport: true,
      },

```
Thank you!

**Comment by chenjiahan** [comment]
For performance reasons, Rspack does not support customName function, see https://github.com/web-infra-dev/rsbuild/issues/3696

**Reply by khangviet1996** [reply]
Thank you so much, I have to refactor it :D
